src/Source/AbstractPropertySourceIterator.php

```diff
@@ -0,0 +1,207 @@
+    }
+
+    public function current()
+    {
+        $current = $this->iterator->current();
+
+        $data = [];
+
+        foreach ($this->fields as $key => $field) {
+            if ($field instanceof ExportField) {
+                $name = $field->getLabel();
+                $propertyPath = $field->getPath();
+                $options = $field->getOptions();
+            } elseif (\is_string($field)) {
+                $name = \is_string($key) ? $key : $field;
+                $propertyPath = $field;
+                $options = [];
+            } else {
+                throw new \TypeError(sprintf(
+                    'Unsupported field type. Field should be either an %s or a string.',
+                    ExportField::class
+                ));
+            }
+
+            try {
+                $propertyValue = $this->propertyAccessor->getValue($current[0], new PropertyPath($propertyPath));
+                $data[$name] = $this->getExportValue($propertyValue, $options);
+            } catch (UnexpectedTypeException $e) {
+                //non existent object in path will be ignored
+                $data[$name] = null;
+            }
+
+            $data[$name] = $this->getExportValue($propertyValue, $options);
+        }
+
+        return $data;
+    }
+
+    public function next(): void
+    {
+        $this->iterator->next();
+    }
+
+    public function key()
+    {
+        return $this->iterator->key();
+    }
+
+    public function valid(): bool
+    {
+        return $this->iterator->valid();
+    }
+
+    abstract function rewind(): void;
+
+    public function setDateTimeFormat(string $dateTimeFormat): void
+    {
+        $this->dateTimeFormat = $dateTimeFormat;
+    }
+
+    public function getDateTimeFormat(): string
+    {
+        return $this->dateTimeFormat;
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @return string|null
+     */
+    protected function getExportValue($value, array $options)
+    {
+        // NEXT_MAJOR: Remove this BC-check.
+        if (method_exists($this, 'getValue')) {
+            return $this->getValue($value);
+        }
+
+        if (isset($options['format_callback'])) {
+            return $options['format_callback']($value, $options);
+        }
+
+        switch (true) {
+            case \is_array($value):
+            case $value instanceof \Traversable:
+                return null;
+            case $value instanceof \DateTimeInterface:
+                return $value->format($options['format'] ?? $this->dateTimeFormat);
+            case $value instanceof \DateInterval:
+                return $this->getDuration($value);
+            case \is_object($value):
+                return method_exists($value, '__toString') ? (string) $value : null;
+            case \is_bool($value):
+                // TODO: trans
+
+                return $value;
+            case \is_string($value):
+                // TODO: translate
+
+                return $value;
```

Comment:
Hi @sonata-project/contributors.

I need some help/advice.
Is there a way to use the symfony translator here ?

Reply:
IMHO you can use `$option['translation_domain']` or other `add` method.

Reply:
Yes, i wanna try to introduce an ExportMapper.

```
public function configureExportFields(ExportMapper $exportMapper)
    {
        $exportMapper->add('foo', CustomListType::class, [
            'custom_param' => ['param1' => 'test' ]
        ]);
        $exportMapper->add('bar', 'trans');
        $exportMapper->add('baz', null, ['label' => 'special_label']);
    }
```

With 
```
public add($field, $type, $options)
{
   $label = $options['label'] ?? $field;
   $export = new ExportField($label, $field, $options)

   // ...
}
```

I gonna update the PR/notes, when the PR is ready.

Reply:
You can add type classes and there translate. Something simmilar to form field.

Reply:
> If yes you can check how list is translate.

My issue is that for the show/list/..., the translator is coming from the twig filter `|trans`, or sometimes from the `$translator` property in the Admin.

Here, in this class, I currently don't have access to a translator property. I was looking for a way to solve this.
The current usage of the QuerySourceIterator is here 
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/src/Model/ModelManager.php#L478
called here
https://github.com/sonata-project/SonataAdminBundle/blob/a0f815bb2a51cdf041956fb595d027b45dcba4c5/src/Admin/AbstractAdmin.php#L669

The current solution I found would be to add methods like
```
setTranslator
setTranslationDomain
```
to the SourceIterator and call them in the AbstractAdmin, when creating the SourceIterator.

And then I could do something like
```
if ($option['trans'] = true) {
    if (null === $this->translator) {
        throw \Exception()
    } else {
        $value = $this->translator->trans($value);
    }
}
```

But I'm not a big fan of this solution, so I was wondering if there was a better way to use the translator in the SourceIterator. @wbloszyk

Reply:
Mybe you dont have to translate it on getValue methods. It will be even better do it while user call export method. For example user will be able generate diffrence  language version based on one time loaded data.